meson.build

```diff
@@ -0,0 +1,81 @@
+project('libzseek', 'c',
+    default_options: ['c_std=c11', 'warning_level=3', 'werror=true'],
+    license: 'BSD-3-Clause',
+    version: '3.2.0')
+add_project_arguments('-D_GNU_SOURCE', language: 'c')
+
+
+threads_dep = dependency('threads')
+zstd_dep = dependency('libzstd', version: '>=1.4.9')
+lz4_dep = dependency('liblz4', version: '>=1.8.3')
+check_dep = dependency('check')
+# From https://mesonbuild.com/howtox.html#add-math-library-lm-portably
+cc = meson.get_compiler('c')
+m_dep = cc.find_library('m', required : false)
+
+libzseek = library('zseek',
+    'src/buffer.c',
+    'src/cache.c',
+    'src/common.c',
+    'src/compress.c',
+    'src/decompress.c',
+    'src/seek_table.c',
+    dependencies: [threads_dep, zstd_dep, lz4_dep],
+    gnu_symbol_visibility: 'hidden',
+    install: true,
+    soversion: '3.2.0')
```

Comment:
The autotools build installs a version of `3.2.0` and a soversion of `3`.

Reply:
Thanks a bunch for pointing this out! It clarified a lot, switching to `version` here and aligning the versions in other places too.

Reply:
Autotools versioning is... fun. (Not.)

I forgot to mention/notice: the autotools build actually creates `3.0.2`, because of how it does the math equation for LIBZSEEK_CURRENT (3), LIBZSEEK_REVISION (2), LIBZSEEK_AGE (2).

What it actually is is `{current - age}.{age}.{revision}` as the version (with `{current - age}` on its own being the soname).

This is terribly confusing and trips people up even when they know about it, clearly.

Reply:
Indeed! Your comment above triggered quite an educating search as it turned out :) Have already aligned the versions (sticking to just manually keeping them in sync for now)

Reply:
Ah, note that project semantic version is not the same as library ABI versioning, though it may sometimes be the same. :)

Reply:
Indeed, that's why I don't use `meson.project_version()` here, although at this stage I 'd consider it too much cognitive overhead to not keep them in sync tbh...

Reply:
BTW, incorporated your feedback in https://github.com/foxeng/libzseek/pull/31. Not in a hurry to merge that, in case you want to have a look :)